geoipupdate/geoip.py

```diff
@@ -1,37 +1,41 @@
 import hashlib
 import os
 import urllib
-import gzip
+import tarfile
 import shutil
 import logging
 
 
-MAXMIND_URL = 'http://geolite.maxmind.com/download/geoip/database/'
-MAXMIND_FILENAME = 'GeoLite2-Country.mmdb.gz'
+license_key = os.environ['LICENSE-KEY']
+MAXMIND_URL = 'https://download.maxmind.com/app/geoip_download?edition_id=GeoLite2-Country' \
+              '&license_key={key}&suffix=tar.gz'.format(key=license_key)
+MAXMIND_SHA256_URL = 'https://download.maxmind.com/app/geoip_download?edition_id=GeoLite2-Country' \
+                     '&license_key={key}&suffix=tar.gz.sha256'.format(key=license_key)
+MAXMIND_FILENAME = 'GeoLite2-Country_20200324.tar.gz'
 LOCAL_FILENAME = 'GeoLite2-Country.mmdb'
 GEOIP_PATH = "/common/static/data/geoip/"
 
 
-def match_md5(fp, md5_url):
+def match_sha256(_file, sha256_url):
     """
 
     Purpose of the method to make sure downloaded file is accurate.
 
     Args:
-         fp (file): downloaded file
-         md5_url (str): url to read md5 hash of file
+         _file (file): downloaded file
+         sha256_url (str): url to read sha256 hash of file
 
     Returns:
         bool: Returns True if downloaded file hash and hash returned by url
         are same otherwise returns False
 
     """
-    md5 = urllib.request.urlopen(md5_url).read()
-    m = hashlib.md5()
-    for line in fp:
-        m.update(line)
-    fp.seek(0)
-    return m.hexdigest() == md5.decode()
+    sha256 = urllib.request.urlopen(sha256_url).read()
+    m = hashlib.sha256()
+    _file = open(_file, 'rb').read()
+    m.update(_file)
```

Comment:
I would suggest `file_content = open(_file, 'rb').read() ` as you are generating hash on file content